packages/tailwindcss/src/functions.test.ts

```diff
@@ -618,3 +620,91 @@ describe('theme function', () => {
     })
   })
 })
+
+describe('in plugins', () => {
+  test('CSS theme functions in plugins are properly evaluated', async () => {
+    let compiled = await compile(
+      css`
+        @plugin "my-plugin";
+
+        @theme reference {
+          --color-red: red;
+          --color-orange: orange;
+          --color-blue: blue;
+          --color-pink: pink;
+        }
+      `,
+      {
+        async loadPlugin() {
+          return plugin(({ addBase }) => {
+            addBase({
+              '.my-base-rule': {
+                color: 'theme(colors.red)',
+                'outline-color': 'theme(colors.orange / 15%)',
+                'background-color': 'theme(--color-blue)',
+                'border-color': 'theme(--color-pink / 10%)',
+              },
+            })
```

Comment:
We should probably also add an `addUtility` to test values inside utilities (since these won't be added to the initial CSS AST so it will be replaced at a different layer.

```suggestion
            addBase({
              '.my-base-rule': {
                color: 'theme(colors.red)',
                'outline-color': 'theme(colors.orange / 15%)',
                'background-color': 'theme(--color-blue)',
                'border-color': 'theme(--color-pink / 10%)',
              },
            })
            
            addUtilities({
              '.my-utility': {
                color: 'theme(colors.red)',
              },
            })
```